File: app/src/main/java/com/example/honoassigment1/FirstFragment.java
package com.example.honoassigment1;

import android.os.Bundle;
import android.util.Log;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.navigation.NavAction;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;
import com.example.honoassigment1.databinding.FragmentFirstBinding;


public class FirstFragment extends Fragment {

    FragmentFirstBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentFirstBinding.inflate(getLayoutInflater());
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getArguments() != null) {
            FirstFragmentArgs args = FirstFragmentArgs.fromBundle(getArguments());
            if (args.getSecondFragmentResult() != null) {
                binding.resultTv.setText(args.getSecondFragmentResult());
                Log.d("Hello", "onViewCreated:" + args.getSecondFragmentResult());
            }
        }
        binding.addBtn.setOnClickListener(v -> {
            int firstNum = Integer.parseInt(binding.edtv1.getText().toString());
            int secondNum = Integer.parseInt(binding.edtv2.getText().toString());
            NavDirections action = FirstFragmentDirections.actionFirstFragmentToSecondFragment(firstNum + secondNum);
            Navigation.findNavController(v).navigate(action);
        });
    }
}

File: app/src/main/java/com/example/honoassignment/Assignment1/FirstFragment.kt
package com.example.honoassignment.Assignment1

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.NavDirections
import androidx.navigation.Navigation.findNavController
import com.example.honoassignment.databinding.FragmentFirstBinding

class FirstFragment : Fragment() {
    var binding: FragmentFirstBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentFirstBinding.inflate(layoutInflater)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (arguments != null) {
            val args = FirstFragmentArgs.fromBundle(requireArguments())
            if (args.secondFragmentResult != null) {
                binding!!.resultTv.text = args.secondFragmentResult
                Log.d("Hello", "onViewCreated:" + args.secondFragmentResult)
            }
        }
        binding!!.addBtn.setOnClickListener { v: View? ->
            val firstNum = binding!!.edtv1.text.toString().toInt()
            val secondNum = binding!!.edtv2.text.toString().toInt()
            val action: NavDirections =
                FirstFragmentDirections.actionFirstFragmentToSecondFragment(firstNum + secondNum)
            findNavController(v!!).navigate(action)
        }
    }
}